Security Risk Assessment for the Firmware update API

athoelke opened the pull request:
This PR adds a new appendix ('D', just before the Change history) to the Firmware update API, providing a Security Risk Assessment for the API. This describes a threat model for the firmware update process, and identifies security requirements that affect the API and its implementation.

There are a few minor changes to the existing specification that relate to the SRA, which are added as preliminary commits. The SRA is a single commit with all of the graphics and text of the appendix.

The SRA is easier to read/review in rendered form, Appendix D is on pages 80-102 of the attached PDF draft: [IHI0093-PSA_Certified_Firmware_Update_API-1.0.0-draft-SRA.pdf](https://github.com/ARM-software/psa-api/files/11490070/IHI0093-PSA_Certified_Firmware_Update_API-1.0.0-draft-SRA.pdf)

Review feedback or comments can be left either with reference to the location in the PDF, or inline with the the source file changes in this PR.

bulislaw reviewed (commented):
This looks great, I like how comprehensive it is and as far as I can tell covering the subject well. Some cosmetic comments.

athoelke commented on doc/fwu/appendix/sra.rst:
Adjusted to left aligned - (getting the caption to move as well is too complex)

athoelke commented on doc/fwu/appendix/sra.rst:
It is not possible to override 'float' settings per image - and without the option of floating it to another page, some of the specs end up with almost empty pages when they include several large diagrams close together...

athoelke commented on doc/fwu/appendix/sra.rst:
I can add that as risk to this description.

athoelke commented on doc/fwu/appendix/sra.rst:
Although that would also mitigate this risk, maintaining separate root keys for every device model, and revision (that cannot share firmware) carries a different set of risks. I would not want to see 'key separation' used as the only/primary mechanism to mitigate this risk.

For comparison, this is a mitigation identified and implemented by other FWU frameworks.

athoelke commented on doc/fwu/appendix/sra.rst:
Fair point. This text was adapted from the SUIT security analysis.

athoelke commented on doc/fwu/appendix/sra.rst:
It would not be difficult to add this to the top of D.4. When there are more than a handful of threats, it might help with navigating to something specific without having to switch to the ToC.

athoelke replied:
I'll include this in the next draft

bulislaw commented:
LGTM.

athoelke commented:
@MarcusJGStreets @bulislaw - I've updated SRA based on a completed review of the SRA. In particular, I have:

* Added new threats: T.DISRUPT_INSTALL and T.FAULT_INJECTION
* Added assumptions: Firmware signing key/infrastructure and Rowhammer are out of scope
* Explicitly stated that verifications must be carried out in trusted components
